tests/test_conversion.py

```diff
@@ -32,3 +32,15 @@ def test_ext():
 def test_concat_type_error():
     with pytest.raises(TypeError):
         m.concat(object(), object())
+
+
+@pytest.mark.parametrize('op', [m.uint, m.sint])
+def test_convert_extend(op):
```

Comment:
did this not come up anywhere else? i guess every usage had `len(T) == n`?

Reply:
Yea, I ran into this bug when doing some code of the riscv_mini, I had to use ext first, then convert.  I think a lot of code might be using `m.UInt[N](val)` rather than using the conversion operator which I think works.